frontend/src/settings/beta/SettingsProfileTab.tsx

```diff
@@ -0,0 +1,95 @@
+import ContentCopy from '@mui/icons-material/ContentCopy'
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import Chip from '@mui/material/Chip'
+import Divider from '@mui/material/Divider'
+import IconButton from '@mui/material/IconButton'
+import Stack from '@mui/material/Stack'
+import { useTheme } from '@mui/material/styles'
+import Tooltip from '@mui/material/Tooltip'
+import Typography from '@mui/material/Typography'
+import { useState } from 'react'
+import { User } from 'types/types'
+
+function SettingsProfileTab({ user }: { user: User }) {
+  const theme = useTheme()
+  const [displayToken, setDisplayToken] = useState(false)
+  const [displayedToken, setDisplayedToken] = useState('')
+
+  const regenerateToken = async () => {
+    const { token } = await fetch('/api/v1/user/token', {
+      method: 'POST',
+    }).then((res) => res.json())
```

Comment:
Shouldn't be using promise chains. You're already using `await`, so could just do:
```
const response = await fetch('/api/v1/user/token', {
  method: 'POST',
})

// error handling is needed here, what if the POST request errors?

return response.json()
``